test/extended/prometheus/prometheus.go

```diff
@@ -503,9 +513,21 @@ var _ = g.Describe("[sig-instrumentation] Prometheus", func() {
 				oc.AdminKubeClient().CoreV1().Pods(ns).Delete(context.Background(), execPod.Name, *metav1.NewDeleteOptions(1))
 			}()
 
+			// Checking Watchdog alert state is done in "should have a Watchdog alert in firing state".
+			allowedAlertNames := []string{
+				"Watchdog",
+				"AlertmanagerReceiversNotConfigured",
+				"PrometheusRemoteWriteDesiredShards",
+			}
+
+			if isTechPreviewCluster(oc) {
```

Comment:
Why do you need this and `allowedFiringAlerts` above?

Reply:
These are two separate test cases that cover slightly different things as far as I can tell. The first excludes these two alerts from its check while this one checks they are firing when they're supposed to IIUC

Reply:
Oh, so it's not `allowedAlertNames`, it's `requiredAlertNames`?  if so, can I get an update?

Reply:
Nope, I got that wrong, the difference is one is in the normal tests and one is in the late tests. They both check that alerts aren't firing apart from some allow list, so the allow is the correct name.

@dgrisonnet Do you know what the concrete difference is between this test `shouldn't report any alerts in firing state apart from Watchdog and AlertmanagerReceiversNotConfigured [Early]` and the late `shouldn't report any alerts in firing or pending state apart from Watchdog and AlertmanagerReceiversNotConfigured and have no gaps in Watchdog firing`?

Reply:
The `Early` one makes sure that any alerts that are supposed to fire instantly, during the cluster initialization are disregarded, whereas the other one disregards alerts that may fire during the e2e tests.
In your case, I suppose that `TechPreviewNoUpgrade` fires instantly if there are any tech preview features enabled in the cluster right? If so, then it is correct to allow the alert in both cases since it can fire at all times.

Reply:
Yep that corrects, it fires instantly and continuously throughout the cluster life as soon as the cluster comes up

Reply:
Everything is good then :+1: